desktop-widgets/divelistview.cpp

```diff
@@ -533,6 +533,12 @@ void DiveListView::selectionChanged(const QItemSelection &selected, const QItemS
 		return;
 	}
 
+	// Avoid recursion: if later we decide to select all dives of a trip,
+	// this function is called again. Avoid this by setting the
+	// programmaticalSelectionChange flag. Note that the recursion never
+	// was a problem, but it just feels correct to avoid this recursion.
+	programmaticalSelectionChange = true;
```

Comment:
we used to have something just like that in our selection code - I tried to find it with some creative `git grep` calls (and search for it in the `git log` to see if/when it was removed, but that failed... so maybe I'm misremembering this.